### DISCUSSION TITLE
Allow override of LokiJsonTextFormatter.Format ?

### DISCUSSION THREAD

**Discussion by tisonv** [serilog-contrib/serilog-sinks-grafana-loki#191] [Ideas]
Hi !

You can inherit from LokiJsonTextFormatter and override pretty much everything but the Format function can not.
Is there a technical reason ?

In one of my project I only need very few of the available `logEvent.Properties` but couldn't add a whitelist to filter them.
I ended using a custom internal class based on `LokiJsonTextFormatter ` but I don't like it since I have to track any changes on the source file.

I didn't notice any problem in Loki / Grafana without all my filtered properties.

Thank you !

**Comment by mishamyte** [comment]
Hi!

It was done historically, prob. Cause it's the only method from the Serilog's `ITextFormatter`. 
It was not a case for a lot of people, but they usually implemented it, when needed a lot of customizations

**Comment by mishamyte** [comment]
Hello @tisonv!

Added `virtual` modifier. Already in [v8.2.0-beta.0](https://www.nuget.org/packages/Serilog.Sinks.Grafana.Loki/8.2.0-beta.0)

**Comment by tisonv** [comment]
Thank you !
May I suggest updating `private readonly JsonValueFormatter _valueFormatter;` to `protected readonly JsonValueFormatter _valueFormatter;` ?

**Reply by mishamyte** [reply]
Oh, damn, missed that. Thank you, will update

**Reply by mishamyte** [reply]
Added in [v8.2.0-beta.1](https://www.nuget.org/packages/Serilog.Sinks.Grafana.Loki/8.2.0-beta.1)
